### PR TITLE
Enhancement/117/load crypto only if necessary

### PR DESCRIPTION
**Pull request by southbite**
Hi guys, the version of happn-crypto-utils has gone up, I made the crypto-utils dependancies more solid, and ensured the utilities work in the browser. The crypto library is only fetched and loaded if the happn server being communicated to is encrypting payloads, this should make the browser client more light weight when it is communicating with a server that is not encrypting payloads.

### CODE REVIEW

File: lib/client/base.js [hunk 1]
```diff
@@ -303,49 +314,53 @@
...
+        });
       }
 
+      _this.serverInfo = serverInfo;
 
-      _this.performRequest(null, 'login', loginParameters, null, function (e, result) {
+      if (_this.serverInfo.encryptPayloads){
 
-        if (e) return done(e);
+        _this.__ensureCryptoLibrary(function(e){
```
Comment (JohanvdWest) [inline lib/client/base.js]: Can this be done with promises. Will make it more readable.
```
_this.__ensureCryptoLibrary()
  .then(doKeySetup)
  .then(doLogin);
```

File: lib/client/base.js [hunk 2]
```diff
@@ -593,6 +608,7 @@
 
   HappnClient.prototype.handle_publication = function (message) {
 
+
```
Comment (JohanvdWest) [inline lib/client/base.js]: Why the extra line?

File: lib/client/base.js (hunk 1 above)
Comment (southbite) [inline lib/client/base.js]: I tried this, but ran into difficulties getting it working on the browser, would rather we had functionality that worked over readability in this case. It also added more code as opposed to less code to the solution.

File: lib/client/base.js (hunk 2 above)
Comment (southbite) [inline lib/client/base.js]: removed line

**Pull request merged**